### PR TITLE
RSDK-9041 Unconditionally output all incoming LogRequests

### PR DESCRIPTION
**Pull request by benjirewis**
RSDK-9041

(cc @stuqdog)

### CODE REVIEW

File: robot/server/server.go [hunk 1]
```diff
@@ -425,9 +425,12 @@ func (s *Server) Log(ctx context.Context, req *pb.LogRequest) (*pb.LogResponse,
...
 	log := req.Logs[0]
 
-	// Use a sublogger of robot logger with correct logger name. Disable caller
-	// to mimic caller passed in from gRPC request.
+	// Use a sublogger of robot logger with correct logger name. Set logger to
+	// DEBUG (most accepting) level: we trust module libraries to handle their
+	// own levels and only send us logs over gRPC that we should be outputting.
+	// Disable caller to mimic caller passed in from gRPC request.
 	logger := s.robot.Logger().Sublogger(log.LoggerName)
+	logger.SetLevel(logging.DEBUG)
```
Comment (benjirewis) [inline robot/server/server.go:433]: The test I wrote (`TestModuleLogging`) still passes without this line. Without this line, the `debug log line` message is _not_ output by the RDK logger, but for some reason the observer still sees it. Does this sound expected @dgottlieb? Is the observer's level not tied to the logger's level?

Comment (dgottlieb) [reply robot/server/server.go:433]: I mean, we'll call "Write" on [all of the appenders or none of them](https://github.com/viamrobotics/rdk/blob/423268bc05232dbeded8f2c3c2ef908581e073b4/logging/impl.go?plain=1#L201-L206). So I don't see why the `NewTestAppender` would not get called, but the `observerCore` would.

File: robot/server/server.go [hunk 2]
```diff
@@ -425,10 +424,15 @@ func (s *Server) Log(ctx context.Context, req *pb.LogRequest) (*pb.LogResponse,
 	}
 	log := req.Logs[0]
 
-	// Use a sublogger of robot logger with correct logger name. Disable caller
-	// to mimic caller passed in from gRPC request.
+	// Use a sublogger of robot logger with correct logger name. Also we "set masquerading" such that:
+	// - The logger level is changed to DEBUG.
+	// - Not add this `server.go` file as a caller to the output log. To mimic the caller passed in
```
Comment (benjirewis) [inline robot/server/server.go:429]:
```suggestion
	// - This `server.go` file is not added as a caller to the output log to mimic the caller passed in
```

**Review by benjirewis — COMMENTED** [review]
Nice.

**Review by dgottlieb — APPROVED** [review]

**Pull request merged**